src/server/supplements/har/harTracer.ts

```diff
@@ -212,6 +212,22 @@ export class HarTracer {
       receive,
     };
     harEntry.time = [dns, connect, ssl, wait, receive].reduce((pre, cur) => cur > 0 ? cur + pre : pre, 0);
+
+    const promise = Promise.all([
+      response.serverIPAddressAndPort().then(server => {
+        if (server?.ipAddress)
+          harEntry.serverIPAddress = server.ipAddress;
+
+        if (server?.port)
+          harEntry._serverPort = server.port;
+      }),
+      response.securityDetails().then(details => {
+        if (details)
+          harEntry._securityDetails = details;
+      }),
+    ]).then(() => {});
+    this._addBarrier(page, promise);
```

Comment:
I refactored it a bit while reading. Feel free to keep your code if you like it. However if it's functionally different to mine let me know because it means I misunderstood something. Thanks!

```suggestion
    this._addBarrier(page, response.serverIPAddressAndPort().then(server => {
      if (server?.ipAddress)
        harEntry.serverIPAddress = server.ipAddress;
      if (server?.port)
        harEntry._serverPort = server.port;
    }));
    this._addBarrier(page, response.securityDetails().then(details => {
      if (details)
        harEntry._securityDetails = details;
    }));
```